docs/standard/serialization/system-text-json/custom-contracts.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Custom serialization and deserialization contracts
+description: "Learn how to write your own contract resolution logic to customize the JSON contract for a type."
+ms.date: 09/26/2022
+---
+# Customize a JSON contract
+
+The <xref:System.Text.Json?displayProperty=fullName> library constructs a JSON *contract* for each .NET type, which defines how the type should be serialized and deserialized. The contract is derived from the type's shape, which includes characteristics such as its properties and fields and whether it implements the <xref:System.Collections.IEnumerable> or <xref:System.Collections.IDictionary> interface. Types are mapped to contracts either at run time using reflection or at compile time using the source generator.
+
+Starting in .NET 7, you can customize these JSON contracts to provide more control over how types are converted into JSON and vice versa. The following list shows just some examples of the types of customizations you can make to serialization and deserialization:
+
+- Serialize private fields and properties.
+- Support multiple names for a single property (for example, if a previous library version used a different name).
+- Ignore properties with a specific name, type, or value.
+- Distinguish between explicit `null` values vs. lack of a value in the JSON payload.
+<!--Add links to blog post when published.-->
+
+## How to opt in
+
+There are two ways to plug into customization. Both involve obtaining a resolver, whose job is to provide a <xref:System.Text.Json.Serialization.Metadata.JsonTypeInfo> instance for each type that needs to be serialized.
+
+- By calling the <xref:System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver.%23ctor> constructor to obtain the <xref:System.Text.Json.JsonSerializerOptions.TypeInfoResolver?displayProperty=nameWithType> and adding your [custom actions](#modifiers) to its <xref:System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver.Modifiers> property.
+
+  For example:
+
+  ```csharp
+  JsonSerializerOptions options = new()
+  {
+      TypeInfoResolver = new DefaultJsonTypeInfoResolver()
+      {
+          Modifiers =
+          {
+              MyCustomModifier1,
+              MyCustomModifier2
+          }
+      }
+  };
+  ```
+
+  If you add multiple modifiers, they'll be called serially.
```

Comment:
Probably doesn't matter much, but it's not often I've seen the term "serially" being used in this context.
```suggestion
  If you add multiple modifiers, they'll be called sequentially.
```